kratos/input_output/logger_table_output.cpp

```diff
@@ -24,20 +24,89 @@
-    for(auto& column_name : mColumnsNames){
-      mColumnsWidth.push_back(column_name.size() + 1);
-      column_name.erase(column_name.find_last_not_of(" ") + 1);
+  LoggerTableOutput::LoggerTableOutput(Parameters rSettings): LoggerOutput(mMyFileStream),
+    mCurrentColumnIndex(0),
+    mColumnsHeaders({}),
+    mColumnsWidth({}),
+    mColumnsTexts({}),
+    mColumnsLabels({}),
+    mFileHeader("")
+    {
+      Parameters default_settings(R"({
+          "file_header" : "",
+          "file_name"   : "",
+          "label"       : "",
+          "columns" : [
+              {
+                  "column_label" : "",
+                  "column_header": ""
+              }
+          ]
+      })");
+
+      rSettings.ValidateAndAssignDefaults(default_settings);
+      std::string filename = rSettings["file_name"].GetString();
+      KRATOS_ERROR_IF(filename=="") << "File not defined for info logger. Please check it is correctly defined" << std::endl;
+      mMyFileStream.open(filename, std::ofstream::trunc); // trunc specifies to out only
+      this->ApplySettings(rSettings);
     }
 
-    const DataCommunicator& r_communicator = ParallelEnvironment::GetDefaultDataCommunicator();
-    if (r_communicator.Rank() == 0) {
-      WriteHeader();
+  LoggerTableOutput::LoggerTableOutput(std::ostream& rMyStream, Parameters rSettings) : LoggerOutput(rMyStream),
+    mCurrentColumnIndex(0),
+    mColumnsHeaders({}),
+    mColumnsWidth({}),
+    mColumnsTexts({}),
+    mColumnsLabels({}),
+    mFileHeader("")
+
+    {
+      Parameters default_settings(R"({
+          "file_header" : "",
+          "label"       : "",
+          "columns" : [
+              {
+                  "column_label" : "",
+                  "column_header": ""
+              }
+          ]
+      })");
+
+      rSettings.ValidateAndAssignDefaults(default_settings);
+      this->ApplySettings(rSettings);
+
     }
-  }
 
-  LoggerTableOutput::LoggerTableOutput(LoggerTableOutput const& Other) : LoggerOutput(Other), mColumnsNames(Other.mColumnsNames), mColumnsWidth(Other.mColumnsWidth) {
-  }
+ void LoggerTableOutput::ApplySettings(Parameters rSettings)
+ {
+      mFileHeader = rSettings["file_header"].GetString();
+      mIdLabel = rSettings["label"].GetString();
+      int ncolumns = rSettings["columns"].size();
+      for(int i=0; i<ncolumns; i++)
+      {
+          std::string coltext = rSettings["columns"][i]["column_header"].GetString();
+          mColumnsHeaders.push_back(coltext);
+          mColumnsLabels.push_back(rSettings["columns"][i]["column_label"].GetString());
+          int tmp = std::max<int>(4,coltext.size());
+          mColumnsWidth.push_back(tmp + 1);
+          mColumnsTexts.push_back("null");
+      }
+      const DataCommunicator& r_communicator = ParallelEnvironment::GetDefaultDataCommunicator();
```

Comment:
hm this should not be used here as it might conflict when some solvers run in MPI but others don't
How do the other loggers handle this? I mean they have some feature for filtering the output no?

Reply:
We check the rank of the message and filter it with [DistributedFilter](https://github.com/KratosMultiphysics/Kratos/blob/199a5ba41d9be269049a8b42a737939a7a564b33/kratos/input_output/logger_message.h#L80)

But I am not sure how to apply it here as there is no message with rank. Maybe, waiting until the first message from root?